src/main/java/com/devotedmc/ExilePearl/listener/PlayerListener.java

```diff
@@ -577,9 +581,17 @@ public void onEntityDeath(EntityDeathEvent e) {
 
 		// These will be priority sorted according to the configured algorithm
 		List<Player> damagers = pearlApi.getDamageLogger().getSortedDamagers(playerId);
-
-		Player killer = null;
-		ExilePearl pearl = null;
+		
+		// Check is player is already exiled
+		if (pearlApi.isPlayerExiled(playerId)) {
+			//Reset bed of exiled player if killer is not null
+			if (!damagers.isEmpty() && pearlApi.getPearlConfig().canPerform(ExileRule.SPAWN_RESET)) {
```

Comment:
This needs to be specific to the type of  pearl the player is held in.

Reply:
Is this not already checked by the isPlayerExiled method?

Reply:
No, "exiled" in this context means just any kind of imprisonment, because originally exiling was the only kind of imprisonment. Any mention of "exile" in the code base might as well be replaced by "prison", the difference is only in the config.